src/main/java/io/anserini/document/tweet/Tweet.java

```diff
@@ -0,0 +1,81 @@
+package io.anserini.document.tweet;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.OptionalDouble;
+import java.util.OptionalLong;
+
+import com.fasterxml.jackson.annotation.JsonSetter;
+import com.fasterxml.jackson.databind.*;
+
+public class Tweet {
+  protected long id;
+  protected String text;
+  protected User user;
+  protected String created_at;
+  protected long retweet_count;
+
+  protected OptionalLong in_reply_to_status_id;
+  protected OptionalLong in_reply_to_user_id;
+  protected Optional<RetweetedStatus> retweeted_status;
+  protected Optional<String> lang;
+  protected Optional<Delete> delete;
+  protected Optional<List<OptionalDouble>> coordinates;
+
+  public long getId() {
+    return id;
+  }
+
+  public String getText() {
+    return text;
+  }
+
+  public User getUser() {
+    return user;
+  }
+
+  public String getCreated_at() {
+    return created_at;
+  }
+
+  public long getRetweet_count() {
+    return retweet_count;
+  }
+
+  @JsonSetter("retweet_count")
+  public void setRetweet_count_internal(JsonNode retweet_count_internal) {
+    if (retweet_count_internal != null) {
+      if (retweet_count_internal.isTextual()) {
+        // retweet_count might say "100+"
+        // TODO: This is ugly, come back and fix later.
+        retweet_count = Long.parseLong(retweet_count_internal.asText().replace("+", ""));
+      } else if (retweet_count_internal.isNumber()) {
+        retweet_count = retweet_count_internal.asLong();
+      }
+    }
+  }
+
+  public OptionalLong getIn_reply_to_status_id() {
```

Comment:
this is pretty ugly... how about we break Java convention and just have method names be the actual tweet fields? e.g., `in_reply_to_status_id`

Getters and setter can just be overloaded?